DigitalLearningSolutions.Data/Services/ActivityService.cs

```diff
@@ -107,7 +108,7 @@ public ReportsFilterOptions GetFilterOptions(int centreId, int? courseCategoryId
             var historicalCourses = courseDataService
                 .GetCoursesEverUsedAtCentreByCategory(centreId, courseCategoryId);
 
-            var courses = availableCourses.Union(historicalCourses)
+            var courses = availableCourses.Union(historicalCourses, new CourseEqualityComparer())

```

Comment:
What's this change about?

Reply:
We're doing a union of Course and CourseAssessmentDetails, which is a subclass. We needed to define an equality operator so that we can remove any matching one, but this needs a custom comparer.